IDR/idr-metadata
# Study parser utility

This PR creates a utility class to parse the content of the version 1 of IDR study files (INI file based).

- for each study file, the mandatory and optional fields are read depending on the study type (experiment vs screen). Errors are thrown on missing mandatory fields
- the metadata block for each study component (experiment, screen) is detected and parsed
- the publications are turned into a list with `Title` and `Authors` as the mandatory fields and `Preprint`, `DOI`, `PMCID`, `PubMed ID` as the optional fields.
- some validation is performed on some fields
- a class is created to generate the attribute of OMERO objects corresponding to the study/experiments/screens (description, name, map annotation key/value pairs)

## Testing

- run `python pyidr/study_parser.py <idrXXX-YYY-ZZZ>/<idrXXX-study.txt>`. This should throw an exception if some invalidities are found
- run `python pyidr/study_parser.py <idrXXX-YYY-ZZZ>/<idrXXX-study.txt> --report`. This should turn the parsed content into fields matching what is on OMERO
- review the output of Travis which should run the parser in non-failing mode

## Next steps

- update the various studies failing the parsing 
- update the script logic and Travis to fail on invalid study metadata
- consume the object classes to assert the correctness of the annotations on IDR (e.g. `python pyidr/study_parser.py --check`)

/cc @pwalczysko @dominikl
